src/lib/ffi/ffi_util.h

```diff
@@ -75,7 +79,24 @@ T& safe_get(botan_struct<T, M>* p) {
    throw FFI_Error("Invalid object pointer", BOTAN_FFI_ERROR_INVALID_OBJECT);
 }
 
-int ffi_guard_thunk(const char* func_name, const std::function<int()>& thunk);
+template <typename T>
```

Comment:
Perhaps restrict this with a concept? `std::invocable<>` should do.

Reply:
Done